yt/visualization/fixed_resolution.py

```diff
@@ -15,13 +15,11 @@
 from yt.utilities.lib.pixelization_routines import pixelize_cylinder
 from yt.utilities.on_demand_imports import _h5py as h5py
 
-from .fixed_resolution_filters import (
-    FixedResolutionBufferFilter,
-    apply_filter,
-    filter_registry,
-)
 from .volume_rendering.api import off_axis_projection
 
+if TYPE_CHECKING:
+    from yt.visualization.fixed_resolution_filters import FixedResolutionBufferFilter
```

Comment:
I am confused about why we need to do this if we're including the class name in quotes in the constructor definition.

Reply:
Ah, good question. Let me clarify this: putting the name between brackets prevents a NameError to be raised at runtime, but it doesn't tell mypy where to look for the actual definition. Since I can't do a normal import at the top level without creating a cycle, I need to use this type-check-only import. This happens very rarely, and I think this is the first time I need it in yt.

Reply:
Thanks!